Fix locating of version when tagging Docker images on Git tag push

eread opened the pull request:
To contribute to https://github.com/lycheeverse/lychee/issues/1833

Fix locating of version when tagging Docker images on Git tag push and also remove tagging major version only, because it would resolve to 'version 0' at the moment.

eread commented on .github/workflows/docker.yml:
```diff
@@ -38,9 +38,8 @@ jobs:
-            type=semver,pattern={{major}}.{{minor}},value=${{ github.ref_name }},prefix=lychee-
-            type=semver,pattern={{major}},value=${{ github.ref_name }},prefix=lychee-
+            type=semver,pattern=lychee-{{version}}
+            type=semver,pattern=lychee-{{major}}.{{minor}}
```
If I understand correctly, `pattern` helps the action locate the version string itself to use to tag the Docker image. I think the fact that `pattern` is just currently the whole tag is why there are build warnings: https://github.com/lycheeverse/lychee/actions/runs/17205362504/job/48804656733#step:3:54.

I think `prefix` would **add** `lychee-` to the version that's found for tagging the Docker image - perhaps that's wanted so that it matches the tag. The last version tagged with a version number was simply tagged `0.15.0` (https://hub.docker.com/layers/lycheeverse/lychee/0.15.0/images/sha256-f71cb370f67fe81eccab4a8b0ff0539c2859f6fd1c80087078922eba07f9dace) and `0.15.0-alpine` (https://hub.docker.com/layers/lycheeverse/lychee/0.15.0-alpine/images/sha256-18cf46741b0a92244c55670895cf7893c75f19f097f469c1274bf6b53c197ece).

eread commented:
@mre Could you review this one against the documentation for `semver` type: https://github.com/marketplace/actions/docker-metadata-action#typesemver?

I'm imaging that if, for example, this change was in place when the `lychee-v0.20.1` tag was pushed:

1. The `type=semver` lines would be triggered, because `type=semver` is used on push tag events.
1. The `pattern=lychee-` settings would help pattern match just the version string, and would return proper semver.
1. By having this action return something valid twice, there would be two additional `--tag` options added to the Docker image build command in a later action.
 
If correct, there would've been, in the end:

 - `lycheeverse/lychee:0.20.1`
 - `lycheeverse/lychee:0.20`
 - `lycheeverse/lychee:0.20.1-alpine`
 - `lycheeverse/lychee:0.20-alpine` 

Does that make sense? Have I missed anything?

mre commented:
Didn't take a look yet, but just posting a comment that this would affect the docs at https://github.com/lycheeverse/lycheeverse.github.io as well. Maybe some other places, too? Can't think of any right now, but if someone depends on the Docker image tagging, then now is the time speak up. 😆

eread commented:
> Didn't take a look yet, but just posting a comment that this would affect the docs at https://github.com/lycheeverse/lycheeverse.github.io as well. Maybe some other places, too? Can't think of any right now, but if someone depends on the Docker image tagging, then now is the time speak up. 😆

@mre Would this look dramatically different to what the project produced before the move to `release-plz` https://hub.docker.com/r/lycheeverse/lychee/tags?name=0.1?

eread commented:
@mre Based on the comment from @aravindan888 here: https://github.com/lycheeverse/lychee/issues/1833#issuecomment-3241130206, I've added back `value=${{ github.ref_name }}`. I'll interpret `value` as being a default value when the tag doesn't match anything. Not sure if we'll ever hit the logic for this, but it does make this a smaller change.

CC @trask 

WDYT @aravindan888?

eread commented:
> Sorry, there's one comment here, which should be addressed before merging. [#1833 (comment)](https://github.com/lycheeverse/lychee/issues/1833#issuecomment-3241130206) Can you take a look?

Thanks for taking a look @mre. I think I addressed that, and mentioned in this comment: https://github.com/lycheeverse/lychee/pull/1834#issuecomment-3243524167. So I've put back `value=${{ github.ref_name }}` compared to an earlier version.

eread commented:
> @mre I've also added another commit: [88c48a2](https://github.com/lycheeverse/lychee/commit/88c48a289bf5753f2f064fe36f695ab57017440c) to fix an error with a job (https://github.com/lycheeverse/lychee/actions/runs/18024185042/job/51287994384) I got when I rebased this PR.
> 
> Not sure what changed, but explicitly specifying `rustfmt` is documented: https://github.com/dtolnay/rust-toolchain?tab=readme-ov-file#inputs.

@mre I've now removed that commit because it's been fixed upstream.

eread commented:
> Looks fine to me. I don't yet know what implications it will have downstream, but we'll see about that. But from my side, this is worth merging to try and fix the Docker tags. Thanks for the work and for your patience, @eread.

Thanks for the merge @mre.
